database/skylink.go

```diff
@@ -0,0 +1,94 @@
+package database
+
+import (
+	"context"
+	"regexp"
+
+	"gitlab.com/NebulousLabs/errors"
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/bson/primitive"
+	"go.mongodb.org/mongo-driver/mongo"
+)
+
+var (
+	skylinkRE *regexp.Regexp = nil
+)
+
+// Skylink represents a skylink object in the DB.
+type Skylink struct {
+	ID      primitive.ObjectID `bson:"_id,omitempty" json:"-"`
+	Skylink string             `bson:"skylink" json:"skylink"`
+	Size    int64              `bson:"size" json:"size"`
+}
+
+// Skylink gets the DB object for the given skylink.
+// If it doesn't exist it creates it.
```

Comment:
Oh I guess this is why the filesize is sometimes 0 and there's no message scheduled. I guess you're not scheduling a message here to have loose coupling, which is nice of course. You could've also put a service on top of it though and have this return `null` in case it's not found and then have the service schedule a metadata fetch + insert a skylink object

Reply:
This is a good idea. I guess we can also scan the table with skylinks for entries with zero size and just process them. We just need to make sure there are no fetches currently queued. Then we can fetch the sizes and update the users who uploaded them. Or we can also skip updating the users on this pass and just recalculate the uploaded amount by each user once per day after we've updated all skylink sizes. That will make sure we have the right counts.

This looks like a good candidate for an independent F/U.

Reply:
F/U: https://github.com/NebulousLabs/skynet-accounts/issues/22